### DISCUSSION TITLE
How to handle spread/variadic argument for a function?

### DISCUSSION THREAD

**Discussion by ekaradon** [colinhacks/zod#1487] [Q&A]
I try to reproduce the following type: 

```typescript
function foo(...parameters: string[]) {
}
```

But I don't see a way to do it with zod. :thinking: 
I've seen this discussion: https://github.com/colinhacks/zod/issues/1282
But I can't make it match the signature from this function here. Is it possible? I know that I could just use a single parameter as an array of string but it make things less readable in my case.

**Comment by JacobWeisenburger** [comment]
I reopened [variadic args function #1282](https://github.com/colinhacks/zod/issues/1282). Hopefully we can get a better answer.

**Comment by TyIsI** [comment]
Just in case anyone stumbles upon this discussion.

`z.function(z.tuple([]).rest(z.string()), zBoolean)` does the trick for me.

![image](https://github.com/user-attachments/assets/1406f3da-91c7-448a-bb71-e4fcbfc3144a)
